org.lflang/src/org/lflang/ast/ToLf.java

```diff
@@ -722,7 +794,8 @@ private String typeAnnotationFor(Type type) {
      * @return A string representation of {@code statementListList}.
      */
     private String indentedStatements(List<EList<? extends EObject>> statementListList, int extraSeparation) {
-        return statementListList.stream().filter(((Predicate<List<? extends EObject>>) List::isEmpty).negate()).map(
+        indentLvl++;
```

Comment:
I am not sure if `indentLvl` really should be a class variable. If it were passed around as a method argument, that would restrict its scope and make it more explicit that its value is affecting the behavior of the method being invoked -- both of which might be nice.

Reply:
While I agree that passing it around as an argument would be nicer, I think it would require large-scale changes (maybe modify LFSwitch so that doSwitch() passes an extra argument for everything so that caseCode() can have access to `indentLvl`?), which I'm not yet confident in making at my current level of familiarity. Any help or suggestions on this is appreciated.

Reply:
After reading your comment, I realized that I do not even understand why the indentation is being tracked at all. If you override `doSwitch` so that wrapping is applied at the very very end, you have the whole string right there, and you know exactly how long all the lines are, including all the indentation. That would also let you insert the line-wrapping logic into one single place instead of three places (`caseCode`, `list`, and `indentedStatements`).

Reply:
The indentation is tracked to take advantage of existing code which breaks the input into indented blocks. If all line-wrapping is done at the end, I don't think there's any way to figure out which line should be indented by how much without redoing the parsing. Unless we cache the indentation levels for each line during the parsing? Then I think it'll be not very different from what I have now.

Reply:
Okay, let's just merge this into `pretty-printer` and move on.